### ISSUE TITLE
Difficulty Setting up Visual Studio with tutorial

### ISSUE THREAD

**Issue by X3n0n3x** [dotnet/docs#23577]
I'm using Visual Studio 2019, and trying to follow the instruction on this page, but keep getting errors that prevent me from building.  I know from this page:
https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tutorials/local-environment
that I should use "File > New > Project" instead of "dotnet new console" and I should use "Debug > Start Without Debugging" instead of "dotnet run", but I still keep getting errors in Visual Studio 2019.

I was able to get things running following the tutorial on this page:
https://docs.microsoft.com/en-us/visualstudio/get-started/csharp/tutorial-console?view=vs-2019
and so I attempted to copy that project and replace the relevant code with the code for:
https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tutorials/local-environment
it still doesn't work.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6272bcb2-80de-570d-3a0c-8887ad008cca
* Version Independent ID: a785589c-ab2e-9b7d-2aaf-3af8895094bf
* Content: [Work with collections - Introduction to C# tutorial](https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tutorials/arrays-and-collections)
* Content Source: [docs/csharp/tour-of-csharp/tutorials/arrays-and-collections.md](https://github.com/dotnet/docs/blob/main/docs/csharp/tour-of-csharp/tutorials/arrays-and-collections.md)
* Product: **dotnet-csharp**
* Technology: **csharp-get-started**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
@X3n0n3x Can you provide more details about what errors you're getting trying to build? That will help us improve this article.

**Comment by X3n0n3x** [comment]
Yes of course.  Apologies, as I am very much completely new to coding / programming and thus may be lacking some of the basic context.

When attempting to follow the instructions in the "A basic list example" section from [this page](https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tutorials/arrays-and-collections), I substitute the various "dotnet" commands with what I understand to be their equivalents from [this article](https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tutorials/local-environment), resulting in a solution with a single project (a C# Console Application) named "list-tutorial".

I replace the entire contents of the Program.cs with the example given in "A basic list example", but that shows the following Error in the error list:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS8400	Feature 'top-level statements' is not available in C# 8.0. Please use language version 9.0 or greater.	list-tutorial	C:\Users\Sean\source\repos\list-tutorial\list-tutorial\Program.cs	4	Active
```

Then trying to use "Debug > Start Without Debugging" to run it just gives a window saying there were build errors.

Next I tried to combine this example with what I had learned in the example to [make a Calculator in Visual Studio](https://docs.microsoft.com/en-us/visualstudio/get-started/csharp/tutorial-console?view=vs-2019), and tried changing the contents of Program.cs to this:

```
using System;
using System.Collections.Generic;

namespace list-tutorial
{
    class Program
    {
        static void Main(string[] args)
        {
        var names = new List<string> { "<name>", "Ana", "Felipe" };
        foreach (var name in names)
        {
            Console.WriteLine($"Hello {name.ToUpper()}!");
        }
        Console.ReadKey();
        }
    }
}
```


But in this case I get the following errors:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1514	{ expected	list-tutorial	C:\Users\Sean\source\repos\list-tutorial\list-tutorial\Program.cs	4	Active
Error	CS0116	A namespace cannot directly contain members such as fields or methods	list-tutorial	C:\Users\Sean\source\repos\list-tutorial\list-tutorial\Program.cs	4	Active
Error	CS1022	Type or namespace definition, or end-of-file expected	list-tutorial	C:\Users\Sean\source\repos\list-tutorial\list-tutorial\Program.cs	5	Active

```
and again get build errors when trying to run it.

I suppose the crux of my issue is that some of the examples are written assuming the user is using .NET CLI, and as I'm using Visual Studio 2019 while it sometimes lets me know what things I need to change, perhaps there are more things I need to be changing that I'm not aware of.

Thank you for your time!

**Comment by BillWagner** [comment]
Thanks for adding those details @X3n0n3x 

On the first try, you would need to change the target version from .NET 3.1 to .NET 5. (Newer updates to Visual Studio let you select .NET 5 when you create the project). Once you've created a project, you can change to .NET 5 using the "Project" -> "Properties" menu.

On the second issue, you are right that `list-tutorial` isn't a valid identifier. C# parses the `-` in the middle as a minus sign (`-`), causing the error.

I'll add this to our backlog to make sure we make it clear to select .NET 5 when you start this tutorial.

**Comment by X3n0n3x** [comment]
Thanks for the info, redoing it with .NET 5 seems to work with [the example code in the original tutorial](https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tutorials/arrays-and-collections)!

When you mention using .NET 5 instead of .NET 3.1, would that apply to some of the other tutorial pages I'd been using recently, such as [this page](https://docs.microsoft.com/en-us/visualstudio/get-started/tutorial-projects-solutions?view=vs-2019) or this [other page](https://docs.microsoft.com/en-us/visualstudio/get-started/csharp/tutorial-console?view=vs-2019) which recommend using .NET 3.1?

In general is it recommended to be using .NET 5 for practicing / learning Visual Studio 2019 / C# unless otherwise stated by a tutorial then?

Thanks again for the clarifications.
